src/main/java/org/klodnicki/service/AccountService.java

```diff
@@ -47,13 +51,98 @@ private List<Account> getAllAccountsFromDatabase() {
     }
 
     public void removeAccount(String pesel) throws NotFoundInDatabaseException {
-        if (!accountExist(pesel)) {
+        if (accountNotExist(pesel)) {
             throw new NotFoundInDatabaseException(Account.class);
         }
         accountRepository.removeAccount(pesel);
     }
 
-    private boolean accountExist(String pesel) {
-        return accountRepository.findAccountByPesel(pesel).isPresent();
+    private boolean accountNotExist(String pesel) {
+        return accountRepository.findAccountByPesel(pesel).isEmpty();
+    }
+
+    public void modifyAccount(String pesel, String parameterToModify, String newInput)
+            throws IllegalArgumentException, NotFoundInDatabaseException {
+
+        if (accountNotExist(pesel)) {
+            throw new NotFoundInDatabaseException(Account.class);
+        }
+
+        if (parameterToModify.equalsIgnoreCase(SortOptionAccount.PESEL.getSortName()) && newInput.length() != 11) {
+            throw new IllegalArgumentException(PESEL_MIN_DIGITS);
+
+        }
+
+        accountRepository.mergeAccount(pesel, prepareParameterToDatabase(parameterToModify), newInput);
+    }
+
+    private String prepareParameterToDatabase(String parameter) throws NotFoundInDatabaseException {
```

Comment:
The thrown exception seems a bit weird. Method doesn't reach repository here so why does it throw NotFoundInDatabaseException? And you even pass Account here like it didn't found Account in database but method isn't even connected with database?

Reply:
you are absolutely right! I changed that so it should work fine now!